src/gossip/service.zig

```diff
@@ -1528,7 +1533,7 @@ pub const GossipService = struct {
 
         // pre-allocate memory to track insertion failures
         var max_inserts_per_push: usize = 0;
-        for (batch_push_messages.items) |*push_message| {
+        for (batch_push_messages.items) |push_message| {
```

Comment:
why dont we want a pointer here? isnt the no-pointer approach an unnecessary stack-copy

Reply:
There will be no stack copy here, the optimizer can easily see that `batch_push_messages.items` or `push_message`  is not modified inside the loop and will index into `batch_push_messages.items`.

The only cases where you don't modify the value of the capture, but need a pointer would be either
- You actually require the pointer itself.
- You're modifying the location where that pointer is inside of the loop, meaning `push_message` loads cannot be memorized as easily.